spark-bigtable_2.12/src/main/scala/com/google/cloud/spark/bigtable/datasources/BigtableTableScanRDD.scala

```diff
@@ -21,30 +21,32 @@ import com.google.cloud.bigtable.data.v2.BigtableDataClient
 import com.google.cloud.bigtable.data.v2.models.Filters.{FILTERS, TimestampFilter}
 import com.google.cloud.bigtable.data.v2.models.Range.ByteStringRange
 import com.google.cloud.bigtable.data.v2.models.{KeyOffset, Query, Row => BigtableRow}
-import com.google.cloud.spark.bigtable._
 import com.google.cloud.spark.bigtable.filters.RowKeyWrapper
 import com.google.common.collect.{BoundType, RangeSet, TreeRangeSet, Range => GuavaRange}
 import com.google.protobuf.ByteString
 import org.apache.spark.rdd.RDD
-import org.apache.spark.{Partition, TaskContext}
+import org.apache.spark.{Partition, SparkContext, TaskContext}
 import org.apache.yetus.audience.InterfaceAudience
 
 import scala.collection.JavaConverters._
 
 @InterfaceAudience.Private
 class BigtableTableScanRDD(
-    relation: BigtableRelation,
     clientKey: BigtableClientKey,
-    filterRangeSet: RangeSet[RowKeyWrapper]
-) extends RDD[BigtableRow](relation.sqlContext.sparkContext, Nil) {
+    filterRangeSet: RangeSet[RowKeyWrapper],
+    tableId: String,
+    sparkContext: SparkContext,
+    startTimestampMicros: Option[Long],
+    endTimestampMicros: Option[Long]
```

Comment:
why changing it from using `BigtableRelation`?

Reply:
The `BigtableRelation` object only gets created when using DataFrames. Originally, we were only creating a `BigtableTableScanRDD` object when using DataFrames (since that was the only data model we were supporting), but now we're also creating a `BigtableTableScanRDD` object when using RDDs, where we don't have a `BigtableRelation` object anymore.